cockatrice/src/spoilerbackgroundupdater.cpp

```diff
@@ -0,0 +1,210 @@
+        }
+
+        /*
+         * ALERT: Ensure two reloads of the card database do not happen
+         * at the same time or a racetime condition can/will happen!
+         */
+        qDebug() << "Spoiler Season Offline, Reloading Database";
+        QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+    }
+    else if (errorCode == QNetworkReply::NoError)
+    {
+        qDebug() << "Spoiler Service Online";
+        startSpoilerDownloadProcess(SPOILERS_URL, true);
+    }
+    else
+    {
+        qDebug() << "Error: Spoiler download failed with reason" << errorCode;
+    }
+}
+
+bool SpoilerBackgroundUpdater::saveDownloadedFile(QByteArray data)
+{
+    QString fileName = settingsCache->getSpoilerCardDatabasePath();
+    QFileInfo fi(fileName);
+    QDir fileDir(fi.path());
+
+    if (!fileDir.exists() && !fileDir.mkpath(fileDir.absolutePath()))
+    {
+        return false;
+    }
+
+    // Check if the data matches. If it does, then spoilers are up to date.
+    if (getHash(fileName) == getHash(data))
+    {
+        if (trayIcon)
+        {
+            trayIcon->showMessage(tr("Spoilers already up to date"), tr("No new spoilers added"));
+        }
+        qDebug() << "Spoilers Up to Date, Reloading Database";
+        QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+        return false;
+    }
+
+    QFile file(fileName);
+    if (!file.open(QIODevice::WriteOnly))
+    {
+        qDebug() << "Spoiler Service Error: File open (w) failed for" << fileName;
+        return false;
+    }
+
+    if (file.write(data) == -1)
+    {
+        qDebug() << "Spoiler Service Error: File write (w) failed for" << fileName;
+        return false;
+    }
+
+    file.close();
+
+    /*
+     * Data written, so reload the card database
+     * ALERT: Ensure two reloads of the card database do not happen
+     * at the same time or a racetime condition can/will happen!
+     */
+    qDebug() << "Spoiler Service Data Written, Reloading Database";
+    QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+
+    // If the user has notifications enabled, let them know
+    // when the database was last updated
+    if (trayIcon)
+    {
+        QList<QByteArray> lines = data.split('\n');
+
+        foreach (QByteArray line, lines)
+        {
+            if (line.indexOf("created:") > -1)
+            {
+                QString timeStamp = QString(line).replace("created:", "").trimmed();
+                trayIcon->showMessage(tr("Spoilers have been updated!"), tr("Last change:") + " " + timeStamp);
+                emit spoilersUpdatedSuccessfully();
+                return true;
+            }
+        }
+    }
+
+    return true;
+}
+
+QByteArray SpoilerBackgroundUpdater::getHash(const QString fileName)
+{
+    QFile file(fileName);
+
+    if (file.open(QFile::ReadOnly))
+    {
+        // Only read the first 512 bytes (enough to get the "created" tag)
+        const QByteArray bytes = file.read(512);
```

Comment:
Might be worth qDebugging this value

Reply:
An empty spoiler.xml file looks like this: https://github.com/Cockatrice/Magic-Spoiler/blob/77ec169ec5ec36352d02db8ab1137a780ba7133e/spoiler.xml

It has 272 bytes, so only checking first 256bytes instead 512 is sufficient?
Adding a do-not-edit hint as requested by you, won't change too much since it could simply be placed behind the timestamp.

Reply:
@tooomm I read the files and put it through systems to get byte counts. Most services tell me i'll need ~400 bytes to be certain to capture the date. As such, I put a power of 2 in place. 512 bytes isn't that much :smile: